extensions/wikia/PortableInfobox/services/PortableInfoboxMobileRenderService.class.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+class PortableInfoboxMobileRenderService extends PortableInfoboxRenderService {
+	const MEDIA_CONTEXT_INFOBOX_HERO_IMAGE = 'infobox-hero-image';
+	const MEDIA_CONTEXT_INFOBOX = 'infobox';
+	const MOBILE_THUMBNAIL_WIDTH = 360;
+	const MINIMAL_HERO_IMG_WIDTH = 300;
+
+	/**
+	 * renders infobox
+	 *
+	 * @param array $infoboxdata
+	 *
+	 * @param $theme
+	 * @param $layout
+	 * @param $accentColor
+	 * @param $accentColorText
+	 * @return string - infobox HTML
+	 */
+	public function renderInfobox( array $infoboxdata, $theme, $layout, $accentColor, $accentColorText ) {
+		$infoboxHtmlContent = '';
+		$heroData = [ ];
+
+		foreach ( $infoboxdata as $item ) {
+			$data = $item['data'];
+			$type = $item['type'];
+
+			if ( $this->isValidHeroDataItem( $item, $heroData ) ) {
+				$heroData[$type] = $data;
+			} elseif ( $this->templateEngine->isSupportedType( $type ) ) {
+				$infoboxHtmlContent .= $this->renderItem( $type, $data );
+			}
+		}
+
+		if ( !empty( $heroData ) ) {
+			$infoboxHtmlContent = $this->renderInfoboxHero( $heroData ) . $infoboxHtmlContent;
+		}
+
+		if ( !empty( $infoboxHtmlContent ) ) {
+			$output = $this->renderItem( 'wrapper', [ 'content' => $infoboxHtmlContent ] );
+		} else {
+			$output = '';
+		}
+
+		\Wikia\PortableInfobox\Helpers\PortableInfoboxDataBag::getInstance()->setFirstInfoboxAlredyRendered( true );
+
+		return $output;
+	}
+
+	protected function renderImage( $data ) {
+		$images = [ ];
+		$count = count( $data );
+		$helper = $this->getImageHelper();
+
+		for ( $i = 0; $i < $count; $i++ ) {
+			$data[$i]['context'] = self::MEDIA_CONTEXT_INFOBOX;
+			$data[$i] = $helper->extendImageData( $data[$i], self::MOBILE_THUMBNAIL_WIDTH );
+
+			if ( !!$data[$i] ) {
+				$images[] = $data[$i];
+			}
+		}
+
+		if ( count( $images ) === 0 ) {
+			return '';
+		}
+
+		// use different template for wikiamobile
+		if ( !$this->isMercury() ) {
+			// always display only the first image on WikiaMobile
+			$data = $images[0];
+			$templateName = 'image-mobile-wikiamobile';
+		} else {
+			if ( count( $images ) === 1 ) {
+				$data = $images[0];
+				$templateName = 'image-mobile';
+			} else {
+				// more than one image means image collection
+				$data = $helper->extendImageCollectionData( $images );
+				$templateName = 'image-collection-mobile';
+			}
+		}
+
+		$data = SanitizerBuilder::createFromType( 'image' )->sanitize( $data );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "image" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)